### PR TITLE
📝 Add backsticks in doc style guide

### PR DESCRIPTION
**Pull request by blegat**
Ref https://github.com/JuliaOpt/JuMP.jl/pull/1800#issuecomment-456648601

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -16,6 +16,10 @@
 # - Example code should be covered by doctests.
 #   - But it's unclear what to do if the code depends on a solver, see
 #     https://github.com/JuliaOpt/JuMP.jl/issues/1175.
+# - When a word is a Julia symbol and not an english word, enclose it with
```
Comment (Peiffap) [inline docs/src/index.md]:
```suggestion
# - When a word is a Julia symbol and not an English word, enclose it with
```

File: docs/src/index.md [hunk 2]
```diff
@@ -16,6 +16,10 @@
...
+#   backsticks and add a `@ref` if it has a docstring in this doc. If it is a
```
Comment (Peiffap) [inline docs/src/index.md]:
```suggestion
#   backticks and add a `@ref` if it has a docstring in this doc. If it is a
```

File: docs/src/index.md [hunk 3]
```diff
@@ -16,6 +16,10 @@
...
+#   plural, add the "s" after the closing backstick. Example:
```
Comment (Peiffap) [inline docs/src/index.md]:
```suggestion
#   plural, add the "s" after the closing backtick. Example:
```

**Comment by Peiffap** [comment]
If you want to avoid having lots of commits for small typo fixes, the feature lets you commit them in batches, iirc.

**Comment by odow** [comment]
Why not have these as part of the proper style guide under a Documentation heading?

**Comment by blegat** [comment]
> Why not have these as part of the proper style guide under a Documentation heading?

I am not opposed to it

**Comment by mlubin** [comment]
> Why not have these as part of the proper style guide under a Documentation heading?

SGTM

**Pull request closed without merging**